spring-integration-core/src/main/java/org/springframework/integration/transformer/FromProtobufTransformer.java

```diff
@@ -0,0 +1,126 @@
+import org.springframework.messaging.Message;
+import org.springframework.messaging.converter.ProtobufMessageConverter;
+import org.springframework.util.Assert;
+import org.springframework.util.ClassUtils;
+
+/**
+ * A Protocol Buffer transformer to instantiate {@link com.google.protobuf.Message} objects from {@code byte[]}.
+ *
+ * @author Christian Tzolov
+ * @since 6.1
+ */
+public class FromProtobufTransformer extends AbstractTransformer implements BeanClassLoaderAware {
+
+	private final ProtobufMessageConverter protobufMessageConverter;
+
+	private final Class<? extends com.google.protobuf.Message> defaultType;
+
+	private ClassLoader beanClassLoader;
+
+	private Expression typeIdExpression = new FunctionExpression<Message<?>>(
+			(message) -> message.getHeaders().get(ProtoHeaders.TYPE));
+
+	private EvaluationContext evaluationContext;
+
+	/**
+	 * Construct an instance with the supplied default type to create.
+	 * @param defaultType the type.
+	 */
+	public FromProtobufTransformer(Class<? extends com.google.protobuf.Message> defaultType) {
+		this(defaultType, new ProtobufMessageConverter());
+	}
+
+	/**
+	 * Construct an instance with the supplied default type and ProtobufMessageConverter instance.
+	 * @param defaultType the type.
+	 * @param protobufMessageConverter the message converter used.
+	 */
+	public FromProtobufTransformer(Class<? extends com.google.protobuf.Message> defaultType,
+			ProtobufMessageConverter protobufMessageConverter) {
+		Assert.notNull(defaultType, "'defaultType' must not be null");
+		this.defaultType = defaultType;
+		this.protobufMessageConverter = protobufMessageConverter;
+	}
+
+	@Override
+	public void setBeanClassLoader(ClassLoader classLoader) {
+		this.beanClassLoader = classLoader;
+	}
+
+	/**
+	 * Set the expression to evaluate against the message to determine the type. Default {@code headers['proto_type']}.
+	 * @param expression the expression.
+	 */
+	public void setTypeExpression(Expression expression) {
+		assertExpressionNotNull(expression);
+		this.typeIdExpression = expression;
+	}
+
+	/**
+	 * Set the expression to evaluate against the message to determine the type id. Default
+	 * {@code headers['proto_type']}.
+	 * @param expression the expression.
+	 */
+	public void setTypeExpressionString(String expression) {
+		assertExpressionNotNull(expression);
+		this.typeIdExpression = EXPRESSION_PARSER.parseExpression(expression);
+	}
+
+	private void assertExpressionNotNull(Object expression) {
+		Assert.notNull(expression, "'expression' must not be null");
+	}
+
+	@Override
+	protected void onInit() {
+		this.evaluationContext = IntegrationContextUtils.getEvaluationContext(getBeanFactory());
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	protected Object doTransform(Message<?> message) {
+
+		Class<? extends com.google.protobuf.Message> targetClass = null;
+		Object value = this.typeIdExpression.getValue(this.evaluationContext, message);
```

Comment:
It would have been nice but apparently the SpEL is not that smart. I'm getting:
"EL1001E: Type conversion problem, cannot convert from java.lang.String to java.lang.Class<?>"